### ISSUE TITLE
Blending modes

### ISSUE THREAD

**Issue by liaprins-czi** [napari/product-heuristics-2020#20]
Curious how the shapes' layer blending modes were chosen? Why wasn't "multiply" included, and others?
__________________________________________________________
**Recommendation**

Consider expanding to more blending modes, e.g. those available in Photoshop (we,, better yet, those available in other imaging software, or go beyond their options if users would want others).

Enhancement
●○○   Low design work
▼▼▽   Medium development work

![blendingmode](https://user-images.githubusercontent.com/53277128/95919468-7cff6300-0d62-11eb-82ca-a379bde67d2a.png)

**Comment by goanpeca** [comment]
@napari/core-devs is there additional context that could be provided on this issue on why they were not added, or some of the development effort involved?

Thanks :)

**Comment by tlambert03** [comment]
Those are the presets that are currently [provided by vispy](https://github.com/vispy/vispy/blob/04d050564b91161b1f8d4949e61c1df3b44af90d/vispy/gloo/wrappers.py#L30-L47).  Adding additional blending modes would require a little research into openGL's [`glBlendFuncSeparate`](https://www.khronos.org/opengl/wiki/GLAPI/glBlendFuncSeparate), and (preferably) an upstream PR.  Good vispy functions to start looking at are [`BaseGlooFunctions.set_state`](https://github.com/vispy/vispy/blob/04d050564b91161b1f8d4949e61c1df3b44af90d/vispy/gloo/wrappers.py#L440) and [`set_blend_func`](https://github.com/vispy/vispy/blob/04d050564b91161b1f8d4949e61c1df3b44af90d/vispy/gloo/wrappers.py#L251)

**Comment by sofroniewn** [comment]
Yes, it's also a little weird that what vispy calls additive isn't strictly additive!! i.e. it doesn't just add and saturate like you'd expect based on https://en.wikipedia.org/wiki/Blend_modes. I'm not sure what we should do about this / if it would be more confusing to change now, or if we need to expose others. I've been tripped up here myself though!

**Comment by sofroniewn** [comment]
I'd probably say that in the absence of further user requests adding more is low priority - i think it's more common to find more blending modes in a graphics editing tool (like a photoshop) and less common to find others in a scientific visualization tool, but if there was a use case in mind that people had for adding more we could add more.

**Comment by liaprins-czi** [comment]
In the Layers tutorial (https://napari.org/tutorials/fundamentals/image.html?highlight=blending%20mode#blending-layers) (and other tutorials) on napari.org we are explaining how each of the layer modes works,  but just at a very high level. For additive it says:
*The final blending mode `additive` will cause the layer to blend with the layers below even when it has full opacity. This mode is especially useful for many cell biology applications where you have multiple different components of a cell labeled in different colors.*

Based on @sofroniewn 's comment above:
> Yes, it's also a little weird that what vispy calls additive isn't strictly additive!! i.e. it doesn't just add and saturate like you'd expect based on https://en.wikipedia.org/wiki/Blend_modes. I'm not sure what we should do about this / if it would be more confusing to change now, or if we need to expose others. I've been tripped up here myself though!

as well as the [image.sc comments](https://forum.image.sc/t/saving-image-from-napari/50379/6), we should consider further explaining in more depth how `additive` blending mode does work in the interim, and perhaps in future update how it works. 

For now it'd be good to update the documentation in the layers tutorial to be a little more thorough, at minimum referencing that the blending modes we have are from [vispy](https://github.com/vispy/vispy/blob/04d050564b91161b1f8d4949e61c1df3b44af90d/vispy/gloo/wrappers.py#L30-L47). I could not find anywhere in the vispy docs explaining how the blending modes work or are intended to work, the closest is that link Talley shared to the actual code, but even linking to that might be more helpful than nothing.
